language/tools/move-cli/src/main.rs

```diff
@@ -107,69 +116,64 @@ enum Command {
 fn maybe_create_dir(dir_name: &str) -> Result<&Path> {
     let dir = Path::new(dir_name);
     if !dir.exists() {
-        fs::create_dir(dir)?
+        fs::create_dir_all(dir)?
     }
     Ok(dir)
 }
 
-/// Compile the user modules in `move_src` and the script in `script_file`
-fn compile(
-    args: &Move,
-    script_file: &Option<String>,
-) -> Result<(OnDiskStateView, Option<CompiledScript>)> {
-    let move_data = maybe_create_dir(&args.move_data)?;
+/// Generate interface files for published files
+fn generate_interface_files(args: &Move) -> Result<()> {
+    move_lang::generate_interface_files(
+        &[args.move_data.clone()],
+        Some(args.move_data.clone()),
```

Comment:
My mental model of `move_data` is "all the stuff that gets stored on-chain". Putting the interface files in `move_data` messes with that model a bit... Would you be amenable to create a separate directory (`generated`? `move_metadata`?) for artifacts like this that don't live on-chain? I expect that we will soon have a lot of other stuff to put in such a directory (e.g., the generated error descriptions, generated docs, generated tx script bindings).

Reply:
Let me move it into the move_build dir. We are already pull in constants from the source language, so another won't hurt